server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -846,13 +846,13 @@ public void moveShards() {
                     final ModelNode sourceNode = nodes.get(shardRouting.currentNodeId());
                     final ModelNode targetNode = nodes.get(moveDecision.getTargetNode().getId());
                     sourceNode.removeShard(shardRouting);
-                    Tuple<ShardRouting, ShardRouting> relocatingShards = routingNodes.relocateShard(
+                    ShardRouting newShard = routingNodes.relocateOrReinitializeShard(
```

Comment:
Possibly we should only do this if we're using the balanced allocator directly: if we're using it to compute the desired balance then I think it makes more sense to simulate a relocation because this will keep the search shards where they are.

Or maybe we can just forbid using the balanced allocator directly when there are unsearchable shards in play. It's deprecated already today.